### PR TITLE
Fix flash messages

### PR DESCRIPTION
**Pull request by AgaDufrat**
The new layout using GOV.UK Design System renders flash notices only if the
keys are "alert" or "notice". Any messages stored under other keys such as
"success" or "danger" were not displayed.

https://trello.com/b/iACGoJN3/govuk-explore-doing

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Review by hannako — CHANGES_REQUESTED** [review]
Great work spotting and implementing a fix for this! 

It might be nice to add links in your commit message to [here](https://github.com/alphagov/collections-publisher/blob/60771a63f39e632c43e2baa5d1ef4abd65d1a926/app/views/layouts/design_system.html.erb#L48) and [here](https://github.com/alphagov/collections-publisher/blob/60771a63f39e632c43e2baa5d1ef4abd65d1a926/app/views/layouts/design_system.html.erb#L54) to show why you've had to make the changes to the key names. 

It also looks like the error_summary partial [needs the error to have](https://github.com/alphagov/collections-publisher/blob/60771a63f39e632c43e2baa5d1ef4abd65d1a926/app/views/shared/_error_summary.html.erb#L8) a `message` attribute. If you try to add a topic with a slug that's already taken it just says `taken` which is the error type. ie add in a message [here](https://github.com/alphagov/collections-publisher/blob/60771a63f39e632c43e2baa5d1ef4abd65d1a926/app/models/tag.rb#L20)

I haven't checked but wonder if there are other validations that are missing and error message attribute too.

**Comment by hannako** [comment]
P.S. you could maybe make a card for this so we track the work as it's not really related to the label change stuff.

**Review by Tetrino — COMMENTED** [review]
👍

**Comment by Tetrino** [comment]
@AgaDufrat I only just caught your comment that you've got more coming, let me know when you're ready :)

**Comment by AgaDufrat** [comment]
I've done that now (see screenshots in https://github.com/alphagov/collections-publisher/pull/1584#issuecomment-1110728312). Ready for review :)

File: spec/features/managing_browse_pages_spec.rb [hunk 1]
```diff
@@ -90,6 +97,8 @@ def and_i_fill_in_the_form
     @content_id = extract_content_id_from(current_path)
   end
 
+  alias_method :and_i_fill_in_the_form_with_already_existing_slug, :and_i_fill_in_the_form
```
Comment (hannako) [inline spec/features/managing_browse_pages_spec.rb:100]: this is cool!

**Review by hannako — APPROVED** [review]
Nice!

**Pull request merged**